modules/swagger-codegen/src/main/resources/Java/pojo.mustache

```diff
@@ -66,7 +66,11 @@ public class {{classname}} {{#parent}}extends {{{parent}}} {{/parent}}{{#parcela
   private {{{datatypeWithEnum}}} {{name}} = {{{defaultValue}}};
   {{/isContainer}}
 
-  {{/vars}}
+  {{/vars}}{{#gson}}{{#discriminator}}
+  public {{classname}}() {
+    this.{{discriminator}} = this.getClass().getSimpleName();
+  }
+  {{/discriminator}}{{/gson}}
```

Comment:
Better do it as follows to avoid empty lines wth spaces:
```
  {{/vars}}
  {{#gson}}
  {{#discriminator}}
......
......
  {{/discriminator}}
  {{/gson}}
```

Reply:
Thanks for the tip. I just realized that swagger codegen uses JMustache...
I've pushed a new commit with this change :)